src/cargo/core/resolver/resolve.rs

```diff
@@ -240,14 +240,11 @@ unable to verify that `{0}` is the same as when the lockfile was generated
         });
         let name = names.next().unwrap_or_else(|| crate_name.clone());
         for n in names {
-            if n == name {
-                continue;
-            }
-            failure::bail!(
-                "multiple dependencies listed for the same crate must \
-                 all have the same name, but the dependency on `{}` \
-                 is listed as having different names",
-                to
+            failure::ensure!(
+                n == name,
+                "The crate `{}` depends on crate `{}` multiple times with different names",
```

Comment:
as a slight nitpick we tend to prefer that error messages start with lowercase letters like the compiler

Reply:
Good to know!